components/org.wso2.carbon.identity.application.authenticator.google/src/main/java/org/wso2/carbon/identity/application/authenticator/google/GoogleOAuth2Authenticator.java

```diff
@@ -58,6 +70,158 @@ private void initTokenEndpoint() {
         }
     }
 
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+
+        boolean canHandle = false;
+        if (log.isDebugEnabled()) {
+            log.debug("Inside OpenIDConnectAuthenticator.canHandle()");
+        }
+        if (StringUtils.isNotBlank(request.getParameter("oneTapEnabled")) &&
+                request.getParameter("oneTapEnabled").equals("YES")) {
+            // Verifying the Cross-Site Request Forgery (CSRF) token.
+            boolean validCookies = validateCSRFCookies(request);
+            if (validCookies) {
+                canHandle = validateJWTFromGOT(request);
+            }
+        } else {
+            canHandle = super.canHandle(request);
+        }
+        return canHandle;
+    }
+
+    /**
+     * This function validated the JWT token sent via Google One Tap respond
+     * @param request
+     * @return
+     */
+    private boolean validateJWTFromGOT(HttpServletRequest request) {
+
+        boolean validCredential;
+        ExternalIdPConfig externalIdPConfig;
+        try {
+            externalIdPConfig = ConfigurationFacade.getInstance()
+                    .getIdPConfigByName("Google Authenticator", "carbon.super");
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        String clientId = FrameworkUtils
+                .getAuthenticatorPropertyMapFromIdP(externalIdPConfig, "GoogleOIDCAuthenticator").get("ClientId");
+
+        String idTokenString = request.getParameter("credential");
+        //Verifying the ID token.
```

Comment:
Shall we fix formatting errors here?

Reply:
Fixed f2aa0c1666f9b5234e7d113125fadd55008081cd